src/PrestaShopBundle/Controller/Admin/Improve/Design/PositionsController.php

```diff
@@ -231,4 +232,35 @@ private function manageLegacyFlashes($messageId)
             );
         }
     }
+
+    /**
+     * Toggle hook status
+     *
+     * @param Request $request
+     *
+     * @return JsonResponse
+     */
+    public function toggleStatusAction(Request $request)
```

Comment:
Would it be possible not to use directly `Hook` class here, but rather create a ToggleHookStatusHandler and use it here ?

You can see in this PR https://github.com/PrestaShop/PrestaShop/pull/12423 an example of how to do it, following ToggleTaxStatusHandler.

This is done in order to comply with our MVC model 😉  this is a migrated Controller, we aim to have no direct call to ObjectModel classes here, but rather to dispatch Commands and Queries to the domain that will be handled at domain layer.

Reply:
Well seen ! I remember doing it here https://github.com/PrestaShop/PrestaShop/pull/18883
I will improve it ... Thanks @matks